app/models/results.js

```diff
@@ -108,92 +108,78 @@ ResultSchema.plugin(QueryPlugin); // install QueryPlugin
 /**
  * Methods
  */
-ResultSchema.pre('validate', function preValidate(next) {
-  let error;
-  const buildSha1 = _.get(this, 'exec.sut.buildSha1');
-
-  const logs = _.get(this, 'exec.logs');
-  if (Array.isArray(logs)) {
-    for (let i = 0; i < logs.length; i += 1) {
-      const file = logs[i];
-      if (!file.name) {
-        error = new Error(`file[${i}].name missing`);
-        break;
-      }
-
-      if (!file.data) {
-        logger.warn(`file[${i}].data missing`);
-        file.data = '';
-        // error = new Error(`file[${i}].data missing`);
-        // break;
-      }
+ResultSchema.methods.getBuildId = function () { // eslint-disable-line func-names
```

Comment:
this looks more like virtual member - buildRef etc

Reply:
hmm - but we doesn't want to give this to API - so maybe this is ok..

Reply:
should it be getBuildRef instead..

Reply:
Would be a better name, agreed.